### PR TITLE
Add error logs for error while sending sms

### PR DESCRIPTION
**Pull request by RushanNanayakkara**

### CODE REVIEW

**Review by nilasini — APPROVED** [review]

**Comment by nilasini** [comment]
Only one test case is failing and the cause for the failure is not related to the above fix hence merging the PR.

 [ERROR] OIDCFederatedIdpInitLogoutTest.testLogout:327 expected:<400> but was:<200>
 [INFO] 
 [ERROR] Tests run: 1926, Failures: 1, Errors: 0, Skipped: 0

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java [hunk 1]
```diff
@@ -1286,6 +1286,9 @@ private boolean getConnection(HttpURLConnection httpConnection, AuthenticationCo
                                 httpResponse);
                     }
                     return true;
+                } else {
+                    log.error("Error while sending SMS: error code is " + httpConnection.getResponseCode()
```
Comment (ShanChathusanda93) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java:1290]: What is the reason that we used `:` here? normally we are using `:` to add a parameter next. Here we can see another text.

Comment (ShanChathusanda93) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java:1290]: `Error while sending SMS. error code ...`

Comment (ShanChathusanda93) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java:1290]: Normally we are printing this type of logs.

Comment (RushanNanayakkara) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java:1290]: This  change will make the log inconsistent with line 1307. Change both?

Comment (ShanChathusanda93) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java:1290]: No need to change. Just needed to clarify.

**Pull request merged**